### PR TITLE
[Fix] Updata `bbox_3d` info in `cam_instances` for `mono3d-det` task

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/datasets/convert_utils.py [hunk 1]
```diff
@@ -253,16 +253,16 @@ def get_kitti_style_2d_boxes(info: dict,
         # transform the center from [0.5, 1.0, 0.5] to [0.5, 0.5, 0.5]
         dst = np.array([0.5, 0.5, 0.5])
         src = np.array([0.5, 1.0, 0.5])
-        loc = loc + dim * (dst - src)
-        loc_3d = np.copy(loc)
-        gt_bbox_3d = np.concatenate([loc, dim, rot], axis=1).astype(np.float32)
+        loc_3d = loc + dim * (dst - src)
```
Comment (ZCMax) [inline mmdet3d/datasets/convert_utils.py:256]: loc_3d -> loc_center

**Review by ZCMax — APPROVED** [review]

File: mmdet3d/datasets/waymo_dataset.py [hunk 2]
```diff
@@ -156,20 +156,12 @@ def parse_ann_info(self, info: dict) -> dict:
...
-
-        else:
-            # in waymo, lidar2cam = R0_rect @ Tr_velo_to_cam
-            # convert gt_bboxes_3d to velodyne coordinates with `lidar2cam`
-            lidar2cam = np.array(
-                info['images'][self.default_cam_key]['lidar2cam'])
-            gt_bboxes_3d = CameraInstance3DBoxes(
-                ann_info['gt_bboxes_3d']).convert_to(self.box_mode_3d,
-                                                     np.linalg.inv(lidar2cam))
+        # in waymo, lidar2cam = R0_rect @ Tr_velo_to_cam
```
Comment (Tai-Wang) [inline mmdet3d/datasets/waymo_dataset.py:159]: Why here remove the if-else?

Comment (Tai-Wang) [reply mmdet3d/datasets/waymo_dataset.py:159]: The behaviors should not be the same?

Comment (Xiangxu-0103) [reply mmdet3d/datasets/waymo_dataset.py:159]: After updating the converter script, `gt_bboxes_3d` in `cam_instances` for `WaymoDataset` and `KittiDataset` will be transformed from `gravity_center` to `bottom_center`.

File: mmdet3d/datasets/convert_utils.py [hunk 3]
```diff
@@ -253,16 +253,16 @@ def get_kitti_style_2d_boxes(info: dict,
         # transform the center from [0.5, 1.0, 0.5] to [0.5, 0.5, 0.5]
         dst = np.array([0.5, 0.5, 0.5])
         src = np.array([0.5, 1.0, 0.5])
-        loc = loc + dim * (dst - src)
-        loc_3d = np.copy(loc)
-        gt_bbox_3d = np.concatenate([loc, dim, rot], axis=1).astype(np.float32)
+        loc_center = loc + dim * (dst - src)
```
Comment (ZwwWayne) [inline mmdet3d/datasets/convert_utils.py:256]: suggest adding some comments in the modification, e.g., here we use gravity center here

File: mmdet3d/datasets/convert_utils.py [hunk 4]
```diff
@@ -292,11 +292,10 @@ def get_kitti_style_2d_boxes(info: dict,
         # If mono3d=True, add 3D annotations in camera coordinates
         if mono3d and (repro_rec is not None):
             repro_rec['bbox_3d'] = np.concatenate(
-                [loc_3d, dim, rot],
-                axis=1).astype(np.float32).squeeze().tolist()
+                [loc, dim, rot], axis=1).astype(np.float32).squeeze().tolist()
```
Comment (ZwwWayne) [inline mmdet3d/datasets/convert_utils.py:295]: suggest adding some comments in the modification, e.g., here we use gravity center here?

File: mmdet3d/datasets/convert_utils.py (hunk 3 above)
Comment (Xiangxu-0103) [reply mmdet3d/datasets/convert_utils.py:257]: comments added

**Pull request merged**